app/src/main/java/fr/free/nrw/commons/customselector/ui/selector/ImageLoader.kt

```diff
@@ -0,0 +1,273 @@
+        return uploadedStatusDao.getUploadedFromImageSHA1(imageSha1)
+    }
+
+    /**
+     * Insert into uploaded status table.
+     */
+    suspend fun insertIntoUploaded(imageSha1:String, modifiedImageSha1:String, imageResult:Boolean, modifiedImageResult: Boolean){
+        uploadedStatusDao.insertUploaded(
+            UploadedStatus(
+                imageSha1,
+                modifiedImageSha1,
+                imageResult,
+                modifiedImageResult
+            )
+        )
+    }
+
+    /**
+     * Get image sha1 from uri, used to retrieve the original image sha1.
+     */
+    suspend fun getImageSHA1(uri: Uri): String {
+        return withContext(ioDispatcher) {
+            mapImageSHA1[uri]?.let{
+                return@withContext it
+            }
+            try {
+                val result = fileUtilsWrapper.getSHA1(context.contentResolver.openInputStream(uri))
+                mapImageSHA1[uri] = result
+                result
+            } catch (e: FileNotFoundException){
+                e.printStackTrace()
+                ""
+            }
+        }
+    }
+
+    /**
+     * Get result data from database.
+     */
+    fun getResultFromUploadedStatus(uploadedStatus: UploadedStatus): Result {
+        if (uploadedStatus.imageResult || uploadedStatus.modifiedImageResult) {
+            return Result.TRUE
+        } else {
+            uploadedStatus.lastUpdated?.let {
+                val duration = Calendar.getInstance().time.time - it.time
+                if (TimeUnit.MILLISECONDS.toDays(duration) < INVALIDATE_DAY_COUNT) {
+                    return Result.FALSE
+                }
+            }
+        }
+        return Result.INVALID
+    }
+
+    /**
+     * Generate Modified SHA1 using present Exif settings.
+     *
+     * @return modified sha1
+     */
+    private suspend fun generateModifiedSHA1(image: Image) : String {
+        return withContext(defaultDispatcher) {
+            val uploadableFile = PickedFiles.pickedExistingPicture(context, image.uri)
+            val exifInterface: ExifInterface? = try {
+                ExifInterface(uploadableFile.file!!)
+            } catch (e: IOException) {
+                Timber.e(e)
+                null
+            }
+            fileProcessor.redactExifTags(exifInterface, fileProcessor.getExifTagsToRedact())
+            val sha1 =
+                fileUtilsWrapper.getSHA1(fileUtilsWrapper.getFileInputStream(uploadableFile.filePath))
+            uploadableFile.file.delete()
+            sha1
+        }
+    }
+
+    /**
+     * CleanUp function.
+     */
+    fun cleanUP() {
+        scope.cancel()
+    }
+
+    /**
+     * Sealed Result class.
+     */
+    sealed class Result {
+        object TRUE : Result()
+        object FALSE : Result()
+        object INVALID : Result()
+        object NOTFOUND : Result()
+        object ERROR : Result()
+    }
+
+    companion object {
+        const val INVALIDATE_DAY_COUNT: Long = 7
```

Comment:
javadoc explaining what this number means, with a simple example.

Reply:
done.